backend-rust/src/graphql_api.rs

```diff
@@ -3346,6 +3559,56 @@ impl SearchResult {
     }
 }
 
+fn decode_value_with_schema(
+    opt_schema: &Option<Type>,
+    schema_name: &str,
+    value: &Vec<u8>,
+    value_name: &str,
+) -> String {
+    match opt_schema {
+        Some(schema) => {
+            let mut cursor = Cursor::new(&value);
+            match schema.to_json(&mut cursor) {
+                Ok(v) => {
+                    to_string(&v).unwrap_or_else(|e| {
```

Comment:
I would not import `serde_json::to_string` like this. It's not clear what kind of string you're turning it into. Use the fully qualified name, then it's obvious you're turning it into JSON.
```suggestion
                    serde_json::to_string(&v).unwrap_or_else(|e| {
```